src/libraries/System.Text.Json/tests/Serialization/CollectionTests/CollectionTests.AsyncEnumerable.cs

```diff
@@ -0,0 +1,320 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Text.Json.Tests.Serialization
+{
+    public static partial class CollectionTests
+    {
+        [Theory]
+        [MemberData(nameof(GetAsyncEnumerableSources))]
+        public static async Task WriteRootLevelAsyncEnumerable<TElement>(IEnumerable<TElement> source, int delayInterval, int bufferSize)
+        {
+            JsonSerializerOptions options = new JsonSerializerOptions
+            {
+                DefaultBufferSize = bufferSize
+            };
+
+            string expectedJson = JsonSerializer.Serialize(source);
+
+            using var stream = new Utf8MemoryStream();
+            var asyncEnumerable = new MockedAsyncEnumerable<TElement>(source, delayInterval);
+            await JsonSerializer.SerializeAsync(stream, asyncEnumerable, options);
+
+            Assert.Equal(expectedJson, stream.ToString());
```

Comment:
Since `TElement` might be a POCO, where there's no guaranteed serialization order, seems we should be using `JsonTestHelper.AssertJsonEqual(expectedJson, stream.ToString())` here.

Reply:
Thanks, wasn't aware of that helper.